dagshub/streaming/filesystem.py

```diff
@@ -80,8 +80,9 @@ def __init__(self,
                 if ismount(self.project_root):
                     raise ValueError('No git project found! (stopped at mountpoint {self.project_root})')
                 self.project_root = self.project_root / '..'
+            self.project_root = Path(os.path.abspath(self.project_root))
```

Comment:
Why not edit the original `self.project_roo = Path('.')`
And why not keep it relative if it's relative, but relative to cwd?

Reply:
Re: keeping it relative
Ok, so there's a function called `_relative_path(path)` which is used at the beginning of all functions. 
I assume its "spec" is:
- If the path is inside of the repo, return its path relative to the repo root
- if the path is outside of the repo (among other edgecases) return None
This path is then used to query the api among other things

It does it by calculating the path's relative path to `self.project_root`
IF `self.project_root` itself is relative, then it suddenly becomes cwd-dependent, and whatever path `self.project_root` points to drifts from the actual project root whenever you change directories.

so, _relative_path basically has this behavior (pseudocode snippet, don't assume it will run):
```python
# cwd = git repo root
fs = DagsHubFilesystem()
# fs.project_root = "."
print(fs._relative_path("data")) # Expected and actual: "data"
os.chdir("data")
print(fs._relative_path("nested-data")) # Expected: "data/nested-data", actually returns "nested-data"
```

This is because of [this line](https://github.com/DagsHub/client/blob/alpha/dagshub/streaming/filesystem.py#L163). Since `fs.project_root = "."`, after you chdir to data, the project_root is still `"."`
So when it gets resolved for relative check, it gets resolved to `<project-root>/data`

Got this snippet working in pycharm so you can see it for yourself
<img width="587" alt="image" src="https://user-images.githubusercontent.com/111061261/193784717-f5a3cb64-c336-44b6-a316-878b5ddcde48.png">

And with this PR:
<img width="587" alt="image" src="https://user-images.githubusercontent.com/111061261/193784955-4f4906c6-c695-4aef-9d5e-9127794bb15f.png">